recipe_scrapers/americastestkitchen.py

```diff
@@ -0,0 +1,106 @@
+# mypy: allow-untyped-defs
+
+import functools
+import json
+import re
+
+from recipe_scrapers._grouping_utils import IngredientGroup
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, normalize_string
+
+
+class AmericasTestKitchen(AbstractScraper):
+
+    @classmethod
+    def host(cls, domain="americastestkitchen.com"):
+        return domain
+
+    def author(self):
+        return self.schema.author()
+
+    def title(self):
+        return self.schema.title()
+
+    def description(self):
+        return self.schema.description()
+
+    def total_time(self):
+        return get_minutes(self._get_additional_details.get("recipeTimeNote"))
+
+    def image(self):
+        return self.schema.image()
+
+    def ingredients(self):
+        ingredients = []
+        for group in self._get_additional_details.get("ingredientGroups"):
+            ingredients += group["fields"]["recipeIngredientItems"]
+        return [self._parse_ingredient_item(i) for i in ingredients]
+
+    def ingredient_groups(self):
+        ingredient_groups = []
+        for group in self._get_additional_details.get("ingredientGroups"):
+            ingredients = [
+                self._parse_ingredient_item(i)
+                for i in group["fields"]["recipeIngredientItems"]
+            ]
+            purpose = group["fields"]["title"]
+            ingredient_groups.append(
+                IngredientGroup(ingredients=ingredients, purpose=purpose)
+            )
+        return ingredient_groups
+
+    def instructions(self):  # add headnote
+        if headnote := self._get_additional_details.get("headnote"):
+            # Ideally this would use HTMLTagStripperPlugin, but I'm not sure how to invoke it here
+            headnote = f"Note: {normalize_string(re.sub(r'<.*?>', '', headnote))}\n"
+        else:
+            headnote = ""
+        return headnote + self.schema.instructions()
+
+    def instructions_group(self):
+        if headnote := self._get_additional_details.get("headnote"):
+            # Ideally this would use HTMLTagStripperPlugin, but I'm not sure how to invoke it here
+            headnote = f"Note: {normalize_string(re.sub(r'<.*?>', '', headnote))}\n"
```

Comment:
I'm attempting to re-use the `HTMLTagStripperPlugin` functionality at the moment; I'll let you know how that goes.

Reply:
We _could_ import some of the Python functions that provide that functionality and then use it directly from the scraper.. but I'm not too keen on that, because the plugin code feels like it should be optional -- making it a dependency of a scraper seems unusual, and we don't do that for any scrapers currently, as far as I can tell.

Reply:
that works for me - is removing that comment the last of the changes that need to take place?

Reply:
Almost!  I think some of the tests may need updating with that :)

Reply:
Oops, I commented in the wrong thread.  I meant that the `if fragment` change will require updating the test JSON.

Reply:
> that works for me - is removing that comment the last of the changes that need to take place?

Let's keep the comment, but rephrase it to something like: we could import <the plugin>, but that would make it -- an optional plugin -- a dependency.